src/shim/support/has.ts

```diff
@@ -128,7 +128,14 @@ add(
 add('es-observable', () => typeof global.Observable !== 'undefined', true);
 
 /* Promise */
-add('es6-promise', () => typeof global.Promise !== 'undefined' && has('es6-symbol'), true);
+add(
+	'es6-promise',
+	() =>
+		typeof global.Promise !== 'undefined' &&
+		has('es6-symbol') &&
+		typeof global.Promise.prototype.finally !== 'undefined',
```

Comment:
I don't think we should couple this to the `es6-promise` sniff, as for example are we not going to override the native on the global in say edge which has no `finally` support but does have `Promise`?

Reply:
The `finally` test should probably be its own test. However, if the promise shim has `finally` and the global one does not, `shim/Promise` either returns the global `Promise` or replaces the global one with our shimmed `Promise` and returns the shimmed one. If we return the global one in Edge, it won't have a `finally` method and will error out at runtime when a user tries to use `finally`. In this case, we have two options: return our shimmed `Promise` as we're doing now, or add `finally` to the global `Promise.prototype`. Which do you recommend?